app/src/main/java/com/sdpteam/connectout/registration/CompleteRegistrationForm.java

```diff
@@ -0,0 +1,102 @@
+package com.sdpteam.connectout.registration;
+
+import static com.sdpteam.connectout.profile.Profile.Gender;
+
+import java.util.Arrays;
+
+import com.sdpteam.connectout.R;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.RadioGroup;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModel;
+import androidx.lifecycle.ViewModelProvider;
+
+public class CompleteRegistrationForm extends Fragment {
+
+    private RegistrationViewModel registrationViewModel;
+    public ViewModelProvider.Factory viewModelFactory; // for testing (mocking)
+
+    public static CompleteRegistrationForm newInstance() {
+        CompleteRegistrationForm completeRegistrationForm = new CompleteRegistrationForm();
+        completeRegistrationForm.viewModelFactory = new ViewModelProvider.Factory() {
+            @NonNull
+            @Override
+            public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
```

Comment:
would you mind explaining why using generic here is useful ?

Reply:
Yes I will write a comment because it is not trivial